### PR TITLE
#issue: 446 - Error alert appears in the payment screen

### PR DESCRIPTION
**Pull request by hani-ibrahim**
working on #446

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/PaymentSelectionViewController.swift [hunk 1]
```diff
@@ -41,28 +41,29 @@ final class PaymentSelectionViewController: UIViewController, UIWebViewDelegate
...
     #endif
 
     private func errorBecuaseRequestCancelled(error: NSError) -> Bool {
         return error.domain == "WebKitErrorDomain"
     }
 
-    private func dismissViewController(result: AtlasResult<Bool>, animated: Bool = true) {
-        navigationController?.popViewControllerAnimated(animated)
-        paymentCompletion?(result)
+    private func showError(error: NSError) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Shouldn't it go to `UserMessage`?

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: but I an do something like `AtlasResult<Bool>.failure(error).process()`

Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Oh, didn't notice it. I was pretty sure you would put all the banner related methods there and maybe drop all UIAlertController.

Comment (bonkey) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: So, shouldn't UserMessage handle all banner issues?

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/PaymentSelectionViewController.swift]: Ah yeah, I worked on the refactor first
ready now

File: AtlasSDK/AtlasSDK/Errors.swift [hunk 2]
```diff
@@ -43,9 +43,11 @@ public enum AtlasAPIError: AtlasErrorType {
 
 public enum AtlasCatalogError: AtlasErrorType {
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/Errors.swift]: Is `Catalog` naming here correct? Don't think so.

File: AtlasUI/AtlasUI/Utils/UserMessage.swift [hunk 3]
```diff
@@ -24,37 +24,33 @@ struct ButtonAction {
...
+    static func clearBannerError() {
+        bannerErrorViewController.hideBanner()
     }
 
-    static func showOK(title title: String) {
-        show(title: title, message: nil, actions: ButtonAction(text: "OK"))
+    static func displayError(error: ErrorType) {
+        let viewController: AtlasUIViewController? = try? Atlas.provide()
+        guard let userPresentable = error as? UserPresentable, atlasUIViewController = viewController else {
+            displayError(AtlasCatalogError.unclassified)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/UserMessage.swift]: Isn't it infinite call?

File: AtlasUI/AtlasUI/de.lproj/Localizable.strings [hunk 4]
```diff
@@ -30,10 +30,12 @@
 "AtlasCatalogError.title" = "Oops";
 "AtlasCatalogError.title.missingAddress" = "No address selected";
 "AtlasCatalogError.title.priceChanged" = "Price changed";
+"AtlasCatalogError.message.unclassified" = "Da ist etwas schief gelaufen.\nWir arbeiten an dem Problem.\nBitte komm später noch einmal vorbei.";
 "AtlasCatalogError.message.outOfStock" = "Leider ist dieser Artikel nicht mehr verfügbar";
 "AtlasCatalogError.message.missingAddress" = "Please select shipping and billing addresses first";
 "AtlasCatalogError.message.priceChanged" = "The price is now %@";
 "AtlasCatalogError.message.paymentMethodNotAvailable" = "Payment method is no longer available";
+"AtlasCatalogError.message.addressInvalid" = "Adresse nicht bekannt, bitte überprüfe sie.";
```
Comment (bonkey) [inline AtlasUI/AtlasUI/de.lproj/Localizable.strings]: If `AtlasCatalogError` is incorrect naming as I think, don't forget about these keys.

File: AtlasSDK/AtlasSDK/Errors.swift (hunk 2 above)
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Errors.swift]: I was thinking about the same thing ... any suggestions?

Comment (bonkey) [inline AtlasSDK/AtlasSDK/Errors.swift]: Isn't it just Checkout?

File: AtlasUI/AtlasUI/Utils/UserMessage.swift (hunk 3 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/UserMessage.swift]: I also thought about it ... but `AtlasCatalogError.unclassified` is `UserPresentable` so it should work. 
I can add the `atlasUIViewController` in another guard for safty

**Comment by hani-ibrahim** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

**Comment by bonkey** [comment]
👍

**Pull request merged**